pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/Metrics.java

```diff
@@ -53,7 +54,7 @@ private Metrics() { // Cannot be instantiated
      *
      * @return The value of the metric, or {@code Double.NaN} if the value couln't be computed
      */
-    public static double get(ClassMetricKey key, ASTAnyTypeDeclaration node) {
+    public static double get(MetricKey<ClassMetric> key, ASTAnyTypeDeclaration node) {
```

Comment:
Hmm I think I got your point. Your strategy may directly address a bigger problem which is the abstraction of this framework into `core` so that similar frameworks can be built for other languages. Such frameworks would ideally have the same structure, but would address different AST nodes, and might represent a system differently (PackageStats is strongly fitted to Java's package system and might not be reusable for other languages as is). Having one static Façade class per language is okay, but ideally should delegate to an instance of a language specific implementation of `Metrics`. Plus you're right that this singleton pattern has caused some problems when testing, because it wasn't reinitialized between each test. This looks like a much robuster solution than what's currenly used 

I'll deal with the abstraction part of the project in a few weeks I think, but I'll keep that in mind, thanks :)